### DISCUSSION TITLE
Metaplasmidspades cli using assembly graph

### DISCUSSION THREAD

**Discussion by pbelmann** [ablab/spades#988] [Q&A] [answered]
Thank you for developing Metaplasmidspades!

I`m using the following command to run Metaplasmidspades and I have two questions.

```
metaplasmidspades.py --assembly-graph test1.gfa --12 test1.fq.gz --only-assembler -k 141 -o out_graph6
```
1)
I would like to reuse the assembly graph build by megahit as input for metaplasmidspades.
The k-mer length used in the graph is 141 which is impossible to provide as input for metaplasmid spades
because it tells that only k-mer length values between 1 and 127 are allowed. Is there any way to change this setting?

2)
Is it intended that one must use `--only-assembler` when using `--assembly-graph`? I did not found this in the documentation.
I'm using this setting after reading the error report that is produced when running metaplasmidspades without `--only-assembler`:
```
== Error ==  to run read error correction, reads should be in FASTQ format (.fq, .fastq, .bam, .fq.gz, .fastq.gz are supported): /test/test1.gfa (single reads, library number: 1, library type: assembly-graph)
```

**Comment by asl** [comment, marked as answer]
> Is it intended that one must use --only-assembler when using --assembly-graph? 

Correct

> because it tells that only k-mer length values between 1 and 127 are allowed. Is there any way to change this setting?

There is no way to do this without recompiling SPAdes from source. Also note that this is completely unsupported configuration, so  unexpected bad stuff might happen
